### PR TITLE
Fixing Public Jira Issue : APIMANAGER-4040

### PR DESCRIPTION
**Pull request by ChamNDeSilva**
Applications are created in Store when "application" and "tier" are empty strings.

### CODE REVIEW

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java [hunk 1]
```diff
@@ -3323,8 +3324,16 @@ public static String jsFunction_addApplication(Context cx,
         String status = null;
         if (args != null && args.length >= 4 && isStringArray(args)) {
             String name = (String) args[0];
+
+            if(StringUtils.isEmpty(name.trim())){
```
Comment (prabathabey) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Fix formatting issues.

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java [hunk 2]
```diff
@@ -3323,8 +3324,16 @@ public static String jsFunction_addApplication(Context cx,
...
+                handleException("Application Name is empty.");
+            }
             String username = (String) args[1];
             String tier = (String) args[2];
+
+            if(StringUtils.isEmpty(tier.trim())){
```
Comment (prabathabey) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Same goes here as well.

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java [hunk 3]
```diff
@@ -3345,6 +3354,8 @@ public static String jsFunction_addApplication(Context cx,
 
             status = apiConsumer.addApplication(application, username);
             return status;
+        } else{
```
Comment (prabathabey) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Some more formatting issues.

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java (hunk 1 above)
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Fixed in https://github.com/wso2/carbon-apimgt/pull/1097

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java (hunk 2 above)
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Fixed in https://github.com/wso2/carbon-apimgt/pull/1097

File: components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java (hunk 3 above)
Comment (ChamNDeSilva) [inline components/apimgt/org.wso2.carbon.apimgt.hostobjects/src/main/java/org/wso2/carbon/apimgt/hostobjects/APIStoreHostObject.java]: Fixed in https://github.com/wso2/carbon-apimgt/pull/1097

**Pull request merged**